client/client.go

```diff
@@ -349,11 +349,14 @@ func (c *simpleHTTPClient) Do(ctx context.Context, act httpAction) (*http.Respon
 
 	select {
 	case <-ctx.Done():
-		err = resp.Body.Close()
-		<-done
-		if err == nil {
-			err = ctx.Err()
+		if err = resp.Body.Close(); err != nil {
```

Comment:
Here is the logic:

1. Client calls cancel or the ctx reaches its deadline.

2. You closes the body, which is a cleanup process. 

Do you think we should report the error encountered in the cleanup process or we should return the error as the original cause? I think the later will meet my expectation. I do not care about the cleanup at all. I just care about it cancels the request.